src/unstract/adapters/x2text/llama_parse/src/llama_parse.py

```diff
@@ -55,6 +57,20 @@ def _call_parser(
         )
 
         try:
+            file_extension = pathlib.Path(input_file_path).suffix
+            if not file_extension:
+                try:
+                    with open(input_file_path, mode="rb") as file_obj:
+                        sample_contents = file_obj.read(100)
+                        file_type = filetype.guess(sample_contents)
```

Comment:
@harini-venkataraman I once remebered @muhammad-ali-e mentioning about python-magic to be used for file extensions - https://pypi.org/project/python-magic/
You could reuse this [function](https://github.com/Zipstack/unstract-adapters/blob/aff1be75e11c0d4389d4d7a3dd93f61177183fbe/src/unstract/adapters/utils.py#L37) directly. See if that helps.

Reply:
@gaya3-zipstack The method suggested return the MIME type. But this case we might need the file extension. Also, each MIME type might be mapped with multiple extensions and it is safe to go with filetype here.